File: sample/todo-app-ios/todo-app-ios/list/TodoAddViewProxy.swift
//
//  TodoAddViewImpl.swift
//  todo-app-ios
//
//  Created by Arkadii Ivanov on 22/02/2020.
//  Copyright © 2020 arkivanov. All rights reserved.
//

import Foundation
import TodoLib

class TodoAddViewProxy : BaseMviView<TodoAddViewModel, TodoAddViewEvent>, TodoAddView, ObservableObject {
    
    @Published var model: TodoAddViewModel?
    
    override func render(model: TodoAddViewModel) {
        self.model = model
    }
}

File: sample/todo-app-js/src/main/kotlin/list/TodoAddViewProxy.kt
package list

import com.arkivanov.mvikotlin.core.view.BaseMviView
import com.arkivanov.mvikotlin.sample.todo.common.view.TodoAddView

class TodoAddViewProxy(
    private val updateState: (TodoAddView.Model) -> Unit
) : BaseMviView<TodoAddView.Model, TodoAddView.Event>(), TodoAddView {

    override fun render(model: TodoAddView.Model) {
        updateState(model)
    }
}
